render-test/runner.cpp

```diff
@@ -33,7 +34,47 @@
 
 using namespace mbgl;
 
-// static 
+// Helper obsever for forwarding map events to the actual implementation
+class MapPassthroughObserver : public MapObserver {
+public:
+    MapPassthroughObserver() : impl(nullptr) {}
+
+    void onDidFailLoadingMap(MapLoadError err, const std::string& msg) override final {
+        if (impl) impl->onDidFailLoadingMap(err, msg);
+    }
+
+    void onDidFinishRenderingMap(RenderMode mode) override final {
+        if (impl) impl->onDidFinishRenderingMap(mode);
+    }
+
+    void onDidBecomeIdle() override final {
+        if (impl) impl->onDidBecomeIdle();
+    }
+
+    void setForwardObserver(MapObserver* observer) { impl = observer; }
+
+private:
+    MapObserver* impl;
+};
+
+class MapBecomeIdleObserver : public MapObserver {
```

Comment:
why do we need both MapBecomeIdleObserver and MapPassthroughObserver ?

Reply:
Good point. There is no need for two different observers